pallets/frequency-tx-payment/src/tests/mock.rs

```diff
@@ -116,7 +116,13 @@ pub type MaximumCapacityBatchLength = ConstU8<10>;
 
 pub struct TestAccountId;
 impl Convert<u64, AccountId> for TestAccountId {
-	fn convert(_x: u64) -> AccountId32 {
+	fn convert(x: u64) -> AccountId32 {
+		// force this key to be an ethereum-compatible key
+		if x == 999 {
```

Comment:
Set the converter to look for a magic number rather than doing numerical gymnastics.